pkg/redact/redact.go

```diff
@@ -48,6 +49,36 @@ func Redact(msg interface{}, val reflect.Value) {
 	}
 }
 
+// Redact - removes data from fields marked as sensitive given a proto message struct
+func RedactSensitiveError(msg interface{}, val reflect.Value, err *error) {
+	// TODO: This needs to be optimized!
+	// Should cache messages that contain no sensitive data to ignore, and should cache the map of tag number to field name
+
+	if !val.IsValid() {
+		return
+	}
+
+	switch val.Kind() {
+	case reflect.Slice:
+		for i := 0; i < val.Len(); i += 1 {
+			RedactSensitiveError(val.Index(i).Interface(), val.Index(i), err)
+		}
+	case reflect.Map:
+		// TODO: Implement map logic. Currently only certificates in identity use it, and are redacted
+	case reflect.Ptr:
+		RedactSensitiveError(msg, val.Elem(), err)
+	case reflect.Struct:
+		redactErrorMessage(msg, val, err)
+	}
+}
+
+// RedactedMessage - returns a copy of a proto message struct with data from fields marked as sensitive redacted
+func RedactedError(msg interface{}, err *error) interface{} {
```

Comment:
This function name is very misleading. It redacts sensitive data from err and doesn't modify msg. Why do we return the cloned message? Why do we even need to clone msg since we are not modifying it? The function name gives the impression that the returned value is an error message with sensitive data redacted, but we are just doing in-place redacting.

Reply:
yeah, I removed clone and return statement. I was about to add a review for it, thanks for pointing that out.